### PR TITLE
Navigation bar fix / Sign out fix

### PR DESCRIPTION
**Pull request by jdtoombs**
Restructure navigation bar, and add sign out functionality to sign out button

### CODE REVIEW

File: app/editor/src/components/header/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { useKeycloakWrapper } from 'tno-core';
```
Comment (jdtoombs) [inline app/editor/src/components/header/Header.tsx:1]: Moved the Header for testing..

Comment (jdtoombs) [reply app/editor/src/components/header/Header.tsx:1]: Will bundle pack up into tno-core

**Review by sturufous — APPROVED** [review]

File: app/editor/src/components/header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <styled.Header {...rest}>
+      <div>
+        <a href="https://www2.gov.bc.ca/gov/content/home">
+          <img alt="BC Gov logo" src={process.env.PUBLIC_URL + '/assets/gov_bc_logo.svg'} />
+        </a>
+      </div>
+      <div>
+        <div className="title">{name}</div>
+        <div className="user">
+          <Button onClick={()=>keycloak.instance.logout()} name="signOut">Sign Out</Button>
```
Comment (Fosol) [inline app/editor/src/components/header/Header.tsx:30]: Does this fix sign out?  It looks like it should.

File: app/editor/src/components/header/HeaderStyled.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    display: flex;
+    flex-direction: row;
+    align-items: stretch;
+    margin-left: 2em;
+    width: 100%;
+
+    .title {
+      flex-grow: 1;
+      font-size: 1.25rem;
+      font-family: BCSans, 'Noto Sans', Verdana, Arial, sans serif;
```
Comment (Fosol) [inline app/editor/src/components/header/HeaderStyled.tsx:31]: We should theme the font-family.

File: app/editor/src/components/navbar/NavBarGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+   * choose a background colour for the navigation bar
+   */
+  backgroundColor?: string;
+  /**
+   * place desired nodes within the navigation bar group
+   */
+  children?: React.ReactNode;
+}
+
+export const NavBarGroup: React.FC<INavBarGroupProps> = ({ children }) => {
```
Comment (Fosol) [inline app/editor/src/components/navbar/NavBarGroup.tsx:14]: Just inherit from the appropriate element to get all normal properties.  Don't do a background color prop, instead use a styled property theme.

File: app/editor/src/components/navbar/NavBarItem.tsx [hunk 5]
```diff
@@ -1,10 +1,24 @@
...
   active?: boolean;
+  /**
+   * the path the item will navigate you to
+   */
+  navigateTo?: string;
 }
 
-export const NavBarItem: React.FC<INavBarItemProps> = ({ children, active }) => {
-  return <styled.NavBarItem>{children}</styled.NavBarItem>;
+export const NavBarItem: React.FC<INavBarItemProps> = ({ children, label, navigateTo }) => {
```
Comment (Fosol) [inline app/editor/src/components/navbar/NavBarItem.tsx:19]: We both need to start adding descriptive comments to our components.

File: app/editor/src/components/navbar/NavBarStyled.tsx [hunk 6]
```diff
@@ -1,12 +1,9 @@
 import styled from 'styled-components';
 
-import { INaVBarProps } from '.';
+import { INavBarGroupProps } from '.';
 
-export const NavBar = styled.div<INaVBarProps>`
+export const NavBarGroup = styled.div<INavBarGroupProps>`
   width: 100%;
   height: 40px;
   background-color: #38598a;
```
Comment (Fosol) [inline app/editor/src/components/navbar/NavBarStyled.tsx:8]: Color should be themed.

File: app/editor/src/components/row/Row.tsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+   * place desired nodes in row
+   */
+  children?: React.ReactNode;
+  /**
+   * additional styling flexibility
+   */
+  style?: CSSProperties;
+}
+
+export const Row: React.FC<IRowProps> = ({ children, style }) => {
```
Comment (Fosol) [inline app/editor/src/components/row/Row.tsx:16]: Just inherit from HTMLDivElement props.  Don't have custom properties interface unless you're adding new features.

**Review by Fosol — COMMENTED** [review]
Just need to fix some lint rules and resolve a few comments.

File: app/editor/src/components/header/Header.tsx (hunk 2 above)
Comment (jdtoombs) [reply app/editor/src/components/header/Header.tsx:30]: Yup it does :)

File: app/editor/src/components/navbar/NavBarItem.tsx (hunk 5 above)
Comment (jdtoombs) [reply app/editor/src/components/navbar/NavBarItem.tsx:19]: I will do that before I merge!

File: app/editor/src/components/navbar/NavBarStyled.tsx (hunk 6 above)
Comment (jdtoombs) [reply app/editor/src/components/navbar/NavBarStyled.tsx:8]: Will adjust

File: app/editor/src/components/col/Col.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { CSSProperties } from 'styled-components';
+
+import * as styled from './ColStyled';
+
+export interface IColProps extends React.ReactHTMLElement<HTMLDivElement> {
+  /**
+   * place desired nodes in row
+   */
+  children?: React.ReactNode;
```
Comment (Fosol) [inline app/editor/src/components/col/Col.tsx:10]: Not required as it's already part of HTMLDivElement

File: app/editor/src/components/col/Col.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  /**
+   * additional styling flexibility
+   */
+  style?: CSSProperties;
```
Comment (Fosol) [inline app/editor/src/components/col/Col.tsx:14]: Not required as it's already part of HTMLDivElement

File: app/editor/src/components/row/Row.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { CSSProperties } from 'styled-components';
+
+import * as styled from './RowStyled';
+
+export interface IRowProps extends React.HTMLAttributes<HTMLDivElement> {
+  /**
+   * place desired nodes in row
+   */
+  children?: React.ReactNode;
```
Comment (Fosol) [inline app/editor/src/components/row/Row.tsx:10]: Not required as it's already part of HTMLDivElement

File: app/editor/src/components/row/Row.tsx [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  /**
+   * additional styling flexibility
+   */
+  style?: CSSProperties;
```
Comment (Fosol) [inline app/editor/src/components/row/Row.tsx:14]: Not required as it's already part of HTMLDivElement

**Review by Fosol — COMMENTED** [review]
Minor change

**Review by Fosol — APPROVED** [review]
Awesome

**Comment by jdtoombs** [comment]
> Awesome

😄 Could you merge for me? Doesn't look like I have the option

**Pull request merged**